sovrin_client/agent/jsonpickle_util.py

```diff
@@ -0,0 +1,64 @@
+import jsonpickle
+
+from anoncreds.protocol.types import PublicKey, RevocationPublicKey, \
+    SecretKey, RevocationSecretKey, AccumulatorSecretKey
+from anoncreds.protocol.utils import toDictWithStrValues, fromDictWithStrValues
+
+DATA_KEY = 'py/integer-element'
+
+
+class CommonIntegerElementHandler(jsonpickle.handlers.BaseHandler):
+    def flatten(self, obj, data):
+        data[DATA_KEY] = toDictWithStrValues(obj)
+        return data
+
+    def restore(self, obj):
+        dict = fromDictWithStrValues(obj[DATA_KEY])
+        return self._restore(dict)
+
+    def _restore(self, dict):
+        raise NotImplemented
+
+
+class PublicKeyHandler(CommonIntegerElementHandler):
```

Comment:
Why don't we use PublicKey.fromStrDict(dict) instead of the lines below (manual creation of a new PublicKey object)?